services/worker/src/worker/job_runners/_datasets_based_job_runner.py

```diff
@@ -34,24 +34,29 @@ def __init__(
         job_info: JobInfo,
         app_config: AppConfig,
         processing_step: ProcessingStep,
-        processing_graph: ProcessingGraph,
         hf_datasets_cache: Path,
     ) -> None:
         super().__init__(
             job_info=job_info,
-            common_config=app_config.common,
-            worker_config=app_config.worker,
+            app_config=app_config,
             processing_step=processing_step,
-            processing_graph=processing_graph,
         )
         self.datasets_based_config = app_config.datasets_based
         self.base_datasets_cache = hf_datasets_cache
 
     def get_cache_subdirectory(self, date: datetime) -> str:
         date_str = date.strftime("%Y-%m-%d-%H-%M-%S")
-        payload = (date_str, self.get_job_type(), self.dataset, self.config, self.split, self.force)
+        # TODO: Refactor, need a way to generate payload based only on provided params
```

Comment:
what is the issue here?

Reply:
I don't like the way we use params in lines 53 - 55, because for dataset job runners we don't need config and split for example.

Reply:
Oh. It's not important, we will just have `None, None,`. The payload is only used for the hash. I think it's OK. We could even put `self.job_info["params"]` to be simpler.